src/Nova-GPU-Driver.md

```diff
@@ -1,62 +1,92 @@
 # Nova GPU Driver
 
-Nova is a driver for GSP-based Nvidia GPUs that is currently under development
-and is being written in Rust.
-
-Currently, the objective is to upstream Rust abstractions for the relevant
-subsystems a prerequisite for the actual driver. Hence, the first mainline
-version of Nova will be a stub driver which helps establishing the necessary
-infrastructure in other subsystems (notably PCI and DRM).
+Nova is a driver for GSP (global system processor) based Nvidia GPUs. It is
+intended to become the successor of Nouveau as the mainline driver for Nvidia
+(GSP) GPUs on Linux.
 
 ## Contact
 
-To contact the team and / or participate in development, please use the mailing
-list: nouveau@lists.freedesktop.org
+Available communication channels are:
+
+- The mailing list: nouveau@lists.freedesktop.org
+- IRC: #nouveau on OFTC
+- [Zulip Chat](https://rust-for-linux.zulipchat.com/#narrow/channel/509436-Nova)
 
 
 ## Resources
 
-- [Official Source Tree](https://gitlab.freedesktop.org/drm/nova)
-- [Announcement E-Mail](https://lore.kernel.org/dri-devel/Zfsj0_tb-0-tNrJy@cassiopeiae/)
+The parts that are already in mainline Linux can be found in
+`drivers/gpu/nova-core/` and `drivers/gpu/drm/nova/`
+
+The out-of-tree repository where active development takes place is located on
+[Freedesktop](https://gitlab.freedesktop.org/drm/nova).
+
+
+## Background
+
+### Why a new driver?
+
+Nouveau was, for the most part, designed for pre-GSP hardware. The driver exists
+since ~2009 and its authors back in the day had to reverse engineer a lot about
+the hardware's internals, resulting in a relatively difficult to maintain
+codebase.
+
+Moreover, Nouveau's maintainers concluded that a new driver, exclusively for
+(new) GSP hardware, would allow for significantly simplifying the driver
+design.
 
-In the source tree, the driver lives in `drivers/gpu/drm/nova`.
 
+### Why write it in Rust?
 
-## Status
+Rust's most desired feature are its guarantees for memory safety, notably the
+elimination of Use-after-Free errors. Those are errors GPU drivers suffer from
+significantly, because GPUs are, by definition, asynchronous in regards to the
+CPU and can handle millions of jobs (i.e., memory objects) simultaneously. Jobs'
```

Comment:
The millions of GPU jobs (which I guess is what you call CUDA or shader "threads" here) are not directly managed by the CPU though. The CPU works at the much higher level of running grids of thread blocks for a given kernel (confusing name, I know :sweat_smile:). So while there is indeed high concurrency, the one mentioned here is AFAICT handled by the hardware.

Reply:
Mentioning the "CPU" was an attempt to highlight that the GPU is basically an asynchronously operating universe, independent from all the CPU tasks

Compared to a dumb bus driver or sth, the GPU can hang, complete jobs sometimes faster, sometimes slower; then there can be N userspace threads waiting for a job to finish, therefore blocking on a conventional kernel file descriptor = CPU code..

I'm open for alternative wordings, but would then ask for a complete example :)